shell/middleware/unauthenticated.js

```diff
@@ -0,0 +1,22 @@
+import { setFavIcon, haveSetFavIcon } from '@shell/utils/favicon';
+import { MANAGEMENT } from '@shell/config/types';
+import { _ALL_IF_AUTHED } from '@shell/plugins/dashboard-store/actions';
+
+export default async function({ store }) {
+  if (haveSetFavIcon()) {
```

Comment:
I was wondering why this gate wasn't internal to `setFavIcon`, makes sense in this context